File: nugurang-back/src/main/java/com/nugurang/entity/BoardEntity.java
package com.nugurang.entity;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.PreRemove;
import javax.persistence.Table;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "board")
public class BoardEntity implements Serializable {
    @Id
    @GeneratedValue
    private Long id;

    @Column(nullable = false, unique = true)
    private String name;

    @OneToMany(mappedBy = "board", cascade = CascadeType.ALL)
    private List<XrefUserBoardEntity> xrefUsers = new ArrayList<>();

    @OneToMany(mappedBy = "board", cascade = CascadeType.ALL)
    private List<ThreadEntity> threads = new ArrayList<>();

    @Builder
    public BoardEntity(String name) {
        this.name = name;
    }
}

File: src/main/kotlin/com/nugurang/entity/BoardEntity.kt
package com.nugurang.entity

import javax.persistence.*

@Entity
@Table(name = "board")
class BoardEntity(
    @Id
    @GeneratedValue
    var id: Long? = null,

    @Column(nullable = false, unique = true)
    var name: String,

    @OneToMany(mappedBy = "board", cascade = [CascadeType.ALL])
    var xrefUsers: MutableList<XrefUserBoardEntity> = mutableListOf()

)